src/App.tsx

```diff
@@ -1,14 +1,64 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+
+export type Select = 'all' | 'active' | 'completed';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [query, setQuery] = useState('');
+  const [selectValue, setSelectValue] = useState<Select>('all');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  const visiableTodos = () => {
+    const filteredTodos = todos;
+
+    if (selectValue !== 'all') {
+      return filteredTodos
+        .filter(todo => {
+          switch (selectValue) {
+            case 'active':
+              return todo.completed !== true;
+
+            case 'completed':
+              return todo.completed === true;
+          }
+        })
+        .filter(todo => {
+          return todo.title.toLowerCase().includes(query.toLowerCase().trim());
+        });
+    }
+
+    return filteredTodos.filter(todo => {
+      return todo.title.toLowerCase().includes(query.toLowerCase().trim());
+    });
+  };
+
+  const handleInput = (value: string) => {
```

Comment:
let's follow the common naming convention for event handlers. pls read [this](https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364) article for more details

```suggestion
  const handleInputChange = (value: string) => {
```